Add document about redistribution terms (#9069)
dotnet/core: license-information.md

# License Information

The .NET projects uses the MIT license for source and binaries.

> ## Third party code

Reviewer: ```suggestion
## Third-party code
```

> The project includes source from other projects, as described in third party notices, for example in the [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/main/THIRD-PARTY-NOTICES.TXT). Only permissively licensed code is allowed, per [project policies](https://github.com/dotnet/runtime/blob/main/CONTRIBUTING.md#copying-files-from-other-projects). 

Reviewer: per guidelines: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/per

```suggestion
The project includes source from other projects, as described in third-party notices, for example in the [dotnet/runtime repo](https://github.com/dotnet/runtime/blob/main/THIRD-PARTY-NOTICES.TXT). Only permissively licensed code is allowed, in accordance with [project policies](https://github.com/dotnet/runtime/blob/main/CONTRIBUTING.md#copying-files-from-other-projects).
```

## Redistribution

> Binaries produced by .NET SDK compilers (C#, F#, VB) can be redistributed without additional restrictions. The only restrictions are based on the license of the compiler inputs used to produce the binary.

Reviewer: the first sentence is very ample but there are other conditions that need to be met in order for that to be true. should we make that more clear in the first sentence?

Author: The word "additional" is intended to clarify that. If that word was not there, it would be very "ample". Fair?

Reviewer: maybe I'm misreading it then. the way I'm interpreting this is that you don't have any restrictions to distribute binaries produced by the compilers. but then, by reading the rest, it seems that is true if all other conditions are met.

Author: It means that the use of the compilers don't by themselves add licensing (or other) terms onto your code. The compilers are completely neutral in that respect. On the other hand, we don't control the myriad ways (nor are we responsible to itemize them) that terms could apply to final binary artifacts or of their use.

If there is a better way to write this, that'd be good. Is the following better?

```suggestion
Binaries produced by .NET SDK compilers (C#, F#, VB) can be redistributed without additional restrictions. However, the  license(s) of compiler inputs may apply terms and restrictions to compiler binary outputs.

```

Author: We discussed this. We'll stick with the existing wording.

> Parts of the .NET runtime embedded in applications, including [platform-specific executable hosts](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#framework-dependent-executable), and [self-contained deployments](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#self-contained-deployment), are subject to [.NET](https://github.com/dotnet/dotnet/blob/main/LICENSE.TXT) and [third party notice](https://github.com/dotnet/dotnet/blob/main/THIRD-PARTY-NOTICES.txt) license terms.

Reviewer: ```suggestion
Parts of the .NET runtime embedded in applications, including [platform-specific executable hosts](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#framework-dependent-executable), and [self-contained deployments](https://learn.microsoft.com/dotnet/core/deploying/deploy-with-cli#self-contained-deployment), are subject to [.NET](https://github.com/dotnet/dotnet/blob/main/LICENSE.TXT) and [third-party notice](https://github.com/dotnet/dotnet/blob/main/THIRD-PARTY-NOTICES.txt) license terms.
```

Binaries that target Windows are subject to [additional terms](license-information-windows.md).

> This document is provided for informative purposes only, and is not itself a license.

Reviewer: ```suggestion
This document is provided for informative purposes only and is not itself a license.
```
